src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,483 @@
+package duke;
+
+import duke.exception.EmptyTaskDescriptionException;
+import duke.exception.MissingDateTimeReferenceException;
+import duke.exception.MissingListIndexException;
+import duke.exception.UndefinedCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+
+public class Duke {
+
+    public static String readTaskDescription(Scanner in) throws EmptyTaskDescriptionException {
```

Comment:
Try to structure your code logically. Your code should read like a story. A way to do this is to declare the methods in the order that they are being called. Since main method is the first point, it can be placed at the top, followed by functions in similar order that they are being executed in the program flow.

Code Quality Guideline: Structure Code Logically